miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -871,4 +871,24 @@ public BoxService getBoxService() {
     return boxService;
   }
 
+  @Override
+  public Sample save(Sample sample) throws IOException {
+    if (sample.getId() == SampleImpl.UNSAVED_ID) {
+      return get(create(sample));
+    } else {
+      update(sample);
+      return get(sample.getId());
+    }
+  }
+
+  
+  @Override
+  public DetailedSample save(DetailedSample sample) throws IOException {
```

Comment:
Why a separate method here for DetailedSample? It looks to be the same as the Sample save, but with casting. If that's the case,what do you think about having your calling context (I assume the WorkflowExecutor or similar) cast the output of `save` to DetailedSample if it needs it to be a DetailedSample?